### PR TITLE
♻️Cleanup some min/max/clamp logic

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/automap.cpp [hunk 1]
```diff
@@ -255,32 +257,14 @@ void SearchAutomapItem(const CelOutputBuffer &out)
...
-	else if (y2 > MAXDUNY)
-		y2 = MAXDUNY;
-
-	for (int i = x1; i < x2; i++) {
-		for (int j = y1; j < y2; j++) {
+	int startX = std::clamp(x - 8, 0, MAXDUNX);
+	int startY = std::clamp(y - 8, 0, MAXDUNY);
+
+	int endX = std::clamp(x + 8, 0, MAXDUNX);
+	int endY = std::clamp(y + 8, 0, MAXDUNY);
```
Comment (glebm) [inline Source/automap.cpp:264]:
```suggestion
	const int startX = std::clamp(x - 8, 0, MAXDUNX);
	const int startY = std::clamp(y - 8, 0, MAXDUNY);

	const int endX = std::clamp(x + 8, 0, MAXDUNX);
	const int endY = std::clamp(y + 8, 0, MAXDUNY);
```

File: Source/qol.cpp [hunk 2]
```diff
@@ -97,9 +99,7 @@ void DrawMonsterHealthBar(const CelOutputBuffer &out)
 	Sint32 yPos = 18;
 	Sint32 border = 3;
 
-	Sint32 maxLife = mon->_mmaxhp;
-	if (mon->_mhitpoints > maxLife)
-		maxLife = mon->_mhitpoints;
+	Sint32 maxLife = std::max(mon->_mmaxhp, mon->_mhitpoints);
```
Comment (glebm) [inline Source/qol.cpp:102]:
```suggestion
	const Sint32 maxLife = std::max(mon->_mmaxhp, mon->_mhitpoints);
```

**Review by glebm — DISMISSED** [review]
Neat!

File: Source/items.cpp [hunk 3]
```diff
@@ -1935,8 +1884,7 @@ void SaveItemPower(int i, item_effect_type power, int param1, int param2, int mi
 		[[fallthrough]];
 	case IPL_DUR_CURSE:
 		items[i]._iMaxDur -= r * items[i]._iMaxDur / 100;
-		if (items[i]._iMaxDur < 1)
-			items[i]._iMaxDur = 1;
+		items[i]._iMaxDur = std::min(items[i]._iMaxDur, 1);
```
Comment (thebigMuh) [inline Source/items.cpp:1887]:
```suggestion
		items[i]._iMaxDur = std::max(items[i]._iMaxDur, 1);

```

File: Source/items.cpp [hunk 4]
```diff
@@ -2092,8 +2040,7 @@ void SaveItemPower(int i, item_effect_type power, int param1, int param2, int mi
 		break;
 	case IPL_FIRERESCLVL:
 		items[i]._iPLFR = 30 - plr[myplr]._pLevel;
-		if (items[i]._iPLFR < 0)
-			items[i]._iPLFR = 0;
+		items[i]._iPLFR = std::min(items[i]._iPLFR, 0);
```
Comment (thebigMuh) [inline Source/items.cpp:2043]:
```suggestion
		items[i]._iPLFR = std::max(items[i]._iPLFR, 0);

```

**Review by thebigMuh — APPROVED** [review]

**Review by julealgon — COMMENTED** [review]
This is such a simple and cool change. Things are starting to almost become readable!

**Comment by AJenbo** [comment]
Thanks, there is likely more, but also I think items.cpp was probably the worst one.

**Comment by AJenbo** [comment]
Got some more big cleanups lined up once I'm done beating i18n into shape.

**Pull request merged**